### PR TITLE
`devfreq` support

### PR DESCRIPTION
**Pull request by dostos**
This PR extends previous cpu utilities to generic processor types.

### Changes:
* Refactoring based on an absl for better error propagation.
* Frequency settor / gettor
* Frequency transition statistics
* Test codes 

Tested on Galaxy S20 & Pixel 6.
### Known issue
Process processor affinity may fail on consecutive execution of the program (same with an old Band code). I checked it works fine once we reboot the machine.

### CODE REVIEW

**Review by widiba03304 — COMMENTED** [review]
LGTM overall!

**Comment by widiba03304** [comment]
I checked that the wireless stat can be fetched by `linux/wireless.h` and it is included in the NDK.
I would work on it after this PR!

File: band/device/util.cc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  if (model == "SM-981N") {
+    return {
+        {DeviceFlag::kDSP, "/sys/class/devfreq/exynos5-busfreq-dmc/cur_freq"},
+        {DeviceFlag::kGPU, "/sys/class/devfreq/exynos5-busfreq-mif/cur_freq"},
+        {DeviceFlag::kNPU, "/sys/class/devfreq/exynos5-busfreq-mif/cur_freq"}};
+  }
+  // other devices
+
+  return std::map<DeviceFlag, std::string>();
+}
```
[the remark this answers is not in the document]
Comment (dostos) [reply band/device/util.cc:84]: Thanks for pointing out. I removed few more duplicate codes moved to the resource monitor.

**Comment by dostos** [comment]
Sounds good to me! I moved directory-related utilities to devices with corresponding implementations for the Windows platform.

**Pull request merged**